sparsity/sparse_frame.py

```diff
@@ -369,12 +373,13 @@ def join(self, other, axis=1, how='outer', level=None):
                 index = np.hstack([self.index, other.index])
                 res = SparseFrame(data, index=index, columns=self._columns)
             else:
-                raise NotImplementedError(
-                    "Joining along axis 0 fails when column names differ."
-                    "This is probably caused by adding all-zeros row.")
-                data, new_index = _matrix_join(self._data.T.tocsr(), other._data.T.tocsr(),
-                                               self._columns, other._columns,
-                                               how=how)
+                data, new_index = _matrix_join(
+                    _append_zero_row(self.data.T.tocsr()),
```

Comment:
appending rows is potentially expensive (memory layout). Why not use `._data` attribute it has the zero row appended already?

Reply:
That's what was wrong with previous implementation. `_data.T` gives us zero-**column** on the right. What we need is `data.T` with zero-row on the bottom. So we either do what I've done here or we re-design `_matrix_join` somehow (I don't have yet any idea how).

Reply:
Ok I see. Then I think this is the best way to go!